daffodil-core/src/main/scala/org/apache/daffodil/dsom/RestrictionUnion.scala

```diff
@@ -73,11 +74,17 @@ final class Restriction(xmlArg: Node, val simpleTypeDef: SimpleTypeDefBase)
 
   lazy val baseQName = {
     val baseQNameNodeSeq = xml \ "@base"
-    val baseQNameString = baseQNameNodeSeq.text
-    val tryBaseQName = QName.resolveRef(baseQNameString, xml.scope,
+    val tryBaseQName = QName.resolveRef(baseQNameNodeSeq.text, xml.scope,
       tunable.unqualifiedPathStepPolicy)
-    Assert.invariant(tryBaseQName.isSuccess)
-    tryBaseQName.get
+    val result = tryBaseQName.isFailure && baseQNameNodeSeq.text.contains(" ") match {
+      case true => {
+        println("Warning: Invalid spaces in base name removed '" + baseQNameNodeSeq.text + "'")
```

Comment:
Also rather than a match case true/false, an if/else is preferable per Scala code style.